encoding/json/encoding_test.go

```diff
@@ -197,6 +197,16 @@ func TestEncodeAddress(t *testing.T) {
 	)
 }
 
+func TestDecodeInvalidAddress(t *testing.T) {
+
+	t.Parallel()
+
+	msg := `{"type":"Address","value":"000000000102030405"}`
+
+	_, err := json.Decode(nil, []byte(msg))
+	require.ErrorContains(t, err, "invalid address prefix: expected `0x`, got 0x3030")
```

Comment:
I see what we're trying to fix, but the new error message has also quite the potential for confusion:
If someone looks just at the error message, it looks like it is wrong – `0x3030` has the prefix `0x`.
Of course, it is the hex encoding of the characters (`'0'` is `0x30`), but the hex-encoding (`0x3030`) in the error message looks as if it was the raw input. 

Maybe indicate that this is the hex-encoding of the value, not the value itself.

Reply:
Thanks, I agree.  I improved it by removing "0x" and encoding both values to hex like this:

`"invalid address prefix: (shown as hex) expected %x, got %x"`